src/ArrayManager.inl

```diff
@@ -119,6 +123,36 @@ void* ArrayManager::reallocate(void* pointer, size_t elems, PointerRecord* point
   return pointer_record->m_pointers[my_space];
 }
 
+#if defined(CHAI_ENABLE_PICK)
+template<typename T>
+CHAI_INLINE
+typename ArrayManager::T_non_const<T> ArrayManager::pick(T* src_ptr, size_t index)
+{
+  auto dst_alloc_strategy = m_resource_manager.getAllocator("HOST").getAllocationStrategy();
+  auto src_alloc_strategy = m_resource_manager.getAllocator(const_cast<T_non_const<T>*>(src_ptr)).getAllocationStrategy();
+  T_non_const<T> dst;
+  m_resource_manager.transfer(&dst /*Host*/,
```

Comment:
Why can't you use the `copy` method here?

Reply:
One pointer in each case (`dst_ptr` in `pick` and `src_ptr` in `set`) is a non-umpire pointer. Hence the call to find the allocation record in the `copy` method would fail.

Reply:
Were you able to update this to remove the `transfer` requirement?